web/modules/mof/src/Form/ModelSubmitForm.php

```diff
@@ -103,6 +103,7 @@ public function form(array $form, FormStateInterface $form_state): array {
             '#title' => $this->t('License path'),
             '#default_value' => $license_data[$id]['license_path'] ?? '',
             '#parents' => ['components', $id, 'license_path'],
+            '#required' => FALSE, // This doen't work :-(
```

Comment:
This doesn't have any effect.

Reply:
The `license_path` field (and `component_path`) are only mandatory if a license is selected, i.e. the license field is not empty for the component.  This is controlled by the #states element in the array.

```
'#states' => [                                                      
    'optional' => [                                                   
        ':input[name="components['.$id.'][license]"]' => ['empty' => TRUE],
    ],                                                                
],  
```

Reply:
Ah! I was actually looking at that line but somehow I thought this was to control the visibility. Silly me. I get it now. I don't know why I didn't just try changing it! :-}
Thanks!